lib/_http_agent.js

```diff
@@ -168,8 +168,10 @@ Agent.prototype.addRequest = function addRequest(req, options, port/*legacy*/,
     // we have a free socket, so use that.
     var socket = this.freeSockets[name].shift();
     // Assign the handle a new asyncId and run any init() hooks.
-    socket._handle.asyncReset();
-    socket[async_id_symbol] = socket._handle.getAsyncId();
+    if (socket._handle) {
```

Comment:
Ack.
There's still the possibility that `createSocket` will be overridden to create sockets that do not have `socket._handle.asyncReset` (this is a method that only `tcp_wrap` and `tls_wrap` have). I should have guarded for that (not simply for the existence of `_handle`), but now I don't think that's an interesting use-case.